src/frontend/staff-portal/src/app/components/jj-dispute-info/jj-count/jj-count.component.ts

```diff
@@ -1,381 +1,445 @@
+
+      if (this.isViewOnly || !this.jjDisputedCount) {
+        this.form.disable();
+      }
+      
+      this.countForm.patchValue(this.jjDisputedCount);
+        
+      if (this.jjDisputedCount.jjDisputedCountRoP) {
+        this.countRoPForm.patchValue(this.jjDisputedCount.jjDisputedCountRoP);
+        this.countRoPForm.patchValue({
+          _jailIntermittent: this.jjDisputedCount.jjDisputedCountRoP.jailIntermittent === this.JailIntermittent.Y,
+          _dismissed: this.jjDisputedCount.jjDisputedCountRoP.dismissed === this.Dismissed.Y,
+          _forWantOfProsecution: this.jjDisputedCount.jjDisputedCountRoP.forWantOfProsecution === this.ForWantOfProsecution.Y,
+          _withdrawn: this.jjDisputedCount.jjDisputedCountRoP.withdrawn === this.Withdrawn.Y,
+          _abatement: this.jjDisputedCount.jjDisputedCountRoP.abatement === this.Abatement.Y
+        });
+      }
+    }
+  }
+
+  private initListeners() {
+    if (this.jjDisputedCount) {
+      // listen for form changes
+      this.form.valueChanges.subscribe(() => {
+        Object.assign(this.jjDisputedCount, this.form.getRawValue()); // get raw value includes disabled fields
+        if (this.jjDisputedCount.latestPleaUpdateTs) {
+          this.jjDisputedCount.latestPleaUpdateTs = new Date(this.jjDisputedCount.latestPleaUpdateTs).toISOString();
+        }
+        this.jjDisputedCount.includesSurcharge = (this.inclSurcharge === "yes" ? this.IncludesSurcharge.Y : this.IncludesSurcharge.N);
+        this.jjDisputedCountUpdate.emit(this.jjDisputedCount);
+      });
+
+      this.countForm.valueChanges.subscribe(() => {
+        this.jjDisputedCount = { ...this.jjDisputedCount, ...this.countForm.value };
+        this.inclSurcharge = (this.jjDisputedCount.includesSurcharge === this.IncludesSurcharge.Y ? "yes" : "no");
+        this.jjDisputedCountUpdate.emit(this.jjDisputedCount);
+      });
+
+      this.countRoPForm.valueChanges.subscribe(() => {
+        this.jjDisputedCount.jjDisputedCountRoP = { ...this.jjDisputedCount.jjDisputedCountRoP, ...this.countRoPForm.value };
+        this.jjDisputedCount.jjDisputedCountRoP.jailIntermittent = this.countRoPForm.get("_jailIntermittent").value ? this.JailIntermittent.Y : this.JailIntermittent.N;
+        this.jjDisputedCount.jjDisputedCountRoP.dismissed = this.countRoPForm.get("_dismissed").value ? this.Dismissed.Y : this.Dismissed.N;
+        this.jjDisputedCount.jjDisputedCountRoP.forWantOfProsecution = this.countRoPForm.get("_forWantOfProsecution").value ? this.ForWantOfProsecution.Y : this.ForWantOfProsecution.N;
+        this.jjDisputedCount.jjDisputedCountRoP.withdrawn = this.countRoPForm.get("_withdrawn").value ? this.Withdrawn.Y : this.Withdrawn.N;
+        this.jjDisputedCount.jjDisputedCountRoP.abatement = this.countRoPForm.get("_abatement").value ? this.Abatement.Y : this.Abatement.N;
+        this.jjDisputedCountUpdate.emit(this.jjDisputedCount);
+      });
+    }
+  }
+
+  ngOnChanges(changes: SimpleChanges): void {

```

Comment:
This method was needed to reload the data after saving. When saving, the parent form reloads the JJDispute, but this component didn't update unless I added this logic.

Reply:
Yes because the url did not change, so need to reload manually. But planning to separate the pages